### ISSUE TITLE
Creatures not spawning the correct biome

### ISSUE THREAD

**Issue by MysticMagpie** [Alex-the-666/Ice_and_Fire#3237]
Hi,
I seem to have encountered an issue where creatures have stopped spawning in their defined biome and I'm not sure why. Like everything spawns everywhere. Ice dragons not in ice biomes, sirens not in ocean biomes and hydra not in swamp biomes. Is this something I can fix in configs? Have i made a mistake? The further away i get from the spawn point the worse it seems to become with more and more modded creatures spawning in one place making it really difficult to move around! Please help!

**Comment by CapitaineYolo** [comment]
I have the same problem.

**Comment by atomicAxolotl** [comment]
Same here, stymphalian birds are absolutely everywhere and making it impossible to play...

**Comment by bessiq** [comment]
MC 1.16.4
Forge 35.1.11
Ice & Fire 2.1.3-4

Same problem. Seems to have something to do with the configs in `iceandfire-biomes.toml`. Leaving that config at defaults is fine, but once you change or add to a mob's spawn biomes, that mob just starts spawning everywhere. I appreciate the work that went into this config - being able to combine registry names, categories, and tags into expressions - but either I am doing it wrong or it's not working as described.

For instance, as a simple test, I added a single additional biome to the deathworm.
default: 	`deathwormBiomes = ["&#overworld+|@desert"]`
modified: `deathwormBiomes = ["&#overworld+|@desert", "minecraft:plains"]`
also tried: `deathwormBiomes = ["minecraft:plains", "&#overworld+|@desert"]`

Wait, I think I figured it out. I thought the pipe | symbol was just for use in expressions with multiple values (like the default ones above). But I put it in the front of the single biome I added and I think it's working properly.
`deathwormBiomes = ["&#overworld+|@desert", "|minecraft:plains"]`

I'm not sure why leaving the | out breaks the mob's spawn rules completely though and causes it to spawn everywhere. But hopefully this helps somebody. Also, this should apply to Alex's Mobs mod too, as it has the same type of biome config.
